packages/lib/gpcircuits/src/list-membership.ts

```diff
@@ -0,0 +1,131 @@
+import { CircuitSignal } from "./types";
+import { BABY_JUB_NEGATIVE_ONE } from "@pcd/util";
+import { PODValue, podValueHash } from "@pcd/pod";
+import { poseidon1, poseidon2, poseidon3, poseidon4} from "poseidon-lite";
+
+const poseidon = [poseidon2, poseidon3, poseidon4];
+
+export type ListMembershipModuleInputs = {
+  valueHash: CircuitSignal;
+  list: CircuitSignal[];
+};
+
+export type ListMembershipModuleInputNamesType = [
+  "valueHash",
+  "list"
+];
+
+export type ListMembershipModuleOutputs = { isMember: CircuitSignal };
+
+export type ListMembershipModuleOutputNamesType = ["isMember"];
+
+export class ListMembership {
+  // TODO: Rework this to allow for other modules that might make use of
```

Comment:
As we discussed, I think separating out the helpers for tuple generation (one at a time) would make this clearer and more reusable.  I'd imagine maybe 3 separate helpers:
- Something to build a single tuple 
- Something to handle chaining together tuples when necessary (this is probably the most complicated/fiddly part to express in config).
- Something to take a bunch of these hashed values and format them for the ListMemebership module.  Mostly dealing with the padding.